fdmi/fol_fdmi_src.c

```diff
@@ -594,6 +598,67 @@ M0_INTERNAL int m0_fol_fdmi_post_record(struct m0_fom *fom)
 	return M0_RC(rc);
 }
 
+M0_INTERNAL bool
+m0_fol_fdmi__filter_kv_substring_match(struct m0_buf  *value,
+                                       const char    **substrings)
+{
+	struct m0_buf s;
+	m0_bcount_t   j;
+	bool          match;
+	int           i;
+
+	for (i = 0; substrings[i] != NULL; ++i) {
+		s = M0_BUF_INIT_CONST(strlen(substrings[i]), substrings[i]);
+		if (value->b_nob < s.b_nob)
+			return false;
+		match = false;
+		/* brute-force */
+		for (j = 0; j <= value->b_nob - s.b_nob; ++j) {
+			if (m0_buf_eq(&s, &M0_BUF_INIT(s.b_nob,
+			                               value->b_addr + j))) {
+				match = true;
+				break;
+			}
+		}
```

Comment:
strstr(3) can be used to find the match?
I think it has more efficient implementation.

Reply:
It would only work with null-terminated strings, and there is no guarantee that all values for CAS kv pairs would be null-terminated.